### PR TITLE
Upgrade to 1.0.0

### PR DESCRIPTION
**Pull request by alexAubin**
## Problem

- Upgrade to 1.0.0
- LDAP should be integrated
- App not integrated with mail

## Solution

-  (c.f. previous PR by yalh : https://github.com/YunoHost-Apps/mobilizon_ynh/pull/58 and https://github.com/YunoHost-Apps/mobilizon_ynh/pull/65 )
- Add ldap conf snippets
- And mail conf snippet

## PR Status
- [X] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [ ] Can be reviewed and tested.

## Package_check results
---

(Dev ci seems to be kinda broken independently of Mobilizon, trying to figure out some other place to run package check...)

[Build Status//badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/mobilizon_ynh%20(aleks)/lastBuild/)

### CODE REVIEW

**Comment by yalh76** [comment]
Partial Own CI results: https://paste.yunohost.org/kiwosubodu.php

Upgrade from previous versions fails because of `(CompileError) config/prod.secret.exs:2: undefined function config/3`

**Comment by yalh76** [comment]
Own CI Test : https://paste.yunohost.org/fefohukuri.php
image

@alexAubin here we are ^^.
In previous PR I was focus on the install script and errors with `instance gen` and when it was solved I forgot the change from the `.env` to `prod.secret.exs`

**Comment by alexAubin** [comment]
@yalh76 : hell yeah thank you so much for the fixes and tests ! :heart: 

The only last thing i'm worring about is

```
if ynh_version_gt "1.0.0" "${previous_version}" ; then
```

in the upgrade script, more precisely because this will return true when comparing "1.0.0" to "1.0.0~ynh1" ... which we don't want to happen ... which was my intention with the previous "hack" using 0.1.1 ... or we could use 0.9 or whatever ?

Other than this, do you think we can yolomerge into master after fixing this ? :eyes:

**Comment by yalh76** [comment]
I think we should use `1.0.0~ynh1` to be more accurate. And yolomerge ^^

I've tested ldap authentication: admin login and password are the ldap/yunohost mail and password of the user choose as admin during install

File: scripts/change_url [hunk 1]
```diff
@@ -110,38 +110,35 @@ fi
...
-ynh_replace_string --match_string="$old_domain" --replace_string="$new_domain" --target_file="$final_path/$app/.env"
-ynh_replace_string --match_string="$old_domain" --replace_string="$new_domain" --target_file="$final_path/$app/config/prod.secret.exs"
+config="$final_path/$app/config/prod.secret.exs"
+ynh_backup_if_checksum_is_different --file="$config"
 
-#=================================================
-# STORE THE CONFIG FILE CHECKSUM
-#=================================================
-ynh_print_info --message="Storing the config file checksum..."
+ynh_replace_string --match_string="$old_domain" --replace_string="$new_domain" --target_file="$config"
```
Comment (alexAubin) [inline scripts/change_url:118]: Ah I realize this is not going to be enough anymore, we also need to tweak the LDAP user that's used for email authentication :/

Comment (yalh76) [reply scripts/change_url:118]: That has to be tested but it's `__YNH_USER__` that is used for authentication https://github.com/YunoHost-Apps/mobilizon_ynh/blob/1fdf878a8d4915bfe01c47c8eb1adbbf2fde3fb6/scripts/install#L170

So maybe, just tweaking the main mail address of the notification user is enough...

**Comment by alexAubin** [comment]
Yolomerging to move forward because many people these last days are still installing the old version, and it seems good enough for merge

**Pull request merged**